### ISSUE TITLE
v2.19.3 issue with many to many generation

### ISSUE THREAD

**Issue by dyardyGIT** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#47]
I have the following tables

Users (UserId, Name)
UserRoles (UserRoleId, UserId, RoleId)

when using the latest generator i am getting the following in the UserRoleConfiguration.cs file (notice the duplicated c.UserId)

   HasRequired(a => a.User).WithMany(b => b.UserRoles).HasForeignKey(c => new { c.UserId, c.UserId }).WillCascadeOnDelete();

Similarly i have 
Roles (RoleId, Name)
RolePermissions (RolePermissionId, RoleId, PermissionId)

in the generated RolePermissionConfiguration.cs i see duplicated c.RoleId

// Foreign keys
            HasRequired(a => a.Role).WithMany(b => b.RolePermissions).HasForeignKey(c => new { c.RoleId, c.RoleId }).WillCascadeOnDelete(); // FK_RolePermissions_Roles

**Comment by Kasper888** [comment]
I'm really waiting for resolving this bug, when it will be fixed???? is there any work around?

**Comment by sjh37** [comment]
Using this SQL:

``` sql
CREATE SCHEMA Issue47
GO
CREATE TABLE Issue47.Users ( UserId int NOT NULL IDENTITY(1,1) PRIMARY KEY, Name VARCHAR(10) NULL )
CREATE TABLE Issue47.[Role] ( RoleId int NOT NULL IDENTITY(1,1) PRIMARY KEY, [Role] VARCHAR(10) NULL )
CREATE TABLE Issue47.UserRoles ( UserRoleId int NOT NULL IDENTITY(1,1) PRIMARY KEY, UserId int NOT NULL, RoleId int NOT NULL )
ALTER TABLE Issue47.UserRoles ADD CONSTRAINT Issue47_UserRoles_userid FOREIGN KEY (UserId) REFERENCES Issue47.Users (UserId)
ALTER TABLE Issue47.UserRoles ADD CONSTRAINT Issue47_UserRoles_roleid FOREIGN KEY (RoleId) REFERENCES Issue47.[Role] (RoleId)
```

The generator created the following:

``` c#
public Issue47_UserRoleConfiguration(string schema)
{
    ToTable("UserRoles", schema);
    ...

    // Foreign keys
    HasRequired(a => a.Issue47_Role).WithMany(b => b.Issue47_UserRoles).HasForeignKey(c => c.RoleId).WillCascadeOnDelete(false); // Issue47_UserRoles_roleid
    HasRequired(a => a.Issue47_User).WithMany(b => b.Issue47_UserRoles).HasForeignKey(c => c.UserId).WillCascadeOnDelete(false); // Issue47_UserRoles_userid
}
```

This looks ok to me.

@dyardyGIT  Could you please provide your DDL SQL which is causing this issue, and I will re-test.

**Comment by sjh37** [comment]
This was solved in case #88
